### PR TITLE
Cleanup stringop.c (partly)

### PR DESCRIPTION
**Pull request by ianyfan**
I've removed a lot of old and unused functions.

I want to refactor some of the other functions. In particular `unescape_string`, and maybe move it to `sway/commands.c` since it is the only place it is used, but I'm not sure about its specification to be confident in rewriting it correctly.

### CODE REVIEW

**Comment by emersion** [comment]
Ping: maybe we should first merge this, and do the remaining work in another PR?

**Comment by ianyfan** [comment]
@emersion I'm fine with that; the branch has been updated and rebased.

File: common/stringop.c [hunk 1]
```diff
@@ -64,12 +64,8 @@ char *lenient_strncat(char *dest, const char *src, size_t len) {
...
 // strcmp that also handles null pointers.
 int lenient_strcmp(char *a, char *b) {
-	if (a == b) {
-		return 0;
-	} else if (!a) {
-		return -1;
-	} else if (!b) {
-		return 1;
+	if (!a || !b || a == b) {
+		return a - b;
```
Comment (emersion) [inline common/stringop.c]: Not a fan of this. This casts a `size_t` into an `int`. (I'm not even sure the cast happens _after_ the subtraction)

Comment (ianyfan) [reply common/stringop.c]: I looked it up, and this is undefined behaviour, so I will revert it.

**Comment by emersion** [comment]
Thanks!

**Pull request merged**